### PR TITLE
ENH: Find events by the run start id

### PR DESCRIPTION
**Pull request by ericdill**
- Related to NSLS-II/dataportal#98
- I'm not at all tied to the name, so thrash away

### CODE REVIEW

**Comment by danielballan** [comment]
Thrashing away: `event` -> `events` because it return a list. Otherwise great.

**Comment by ericdill** [comment]
I'm matching the existing API: https://github.com/NSLS-II/metadatastore/pull/107/files#diff-86f2983c849d2ca24f0c14643a623d33R482

File: metadatastore/commands.py [hunk 1]
```diff
@@ -495,7 +493,25 @@ def find_event(run_start):
...
+    return find_event_by_runstart_id(run_start.id)
+
+
+@_ensure_connection
+def find_event_by_runstart_id(run_start_id):
+    """Returns a set of events given a RunStart object
+
+    Parameters
+    ---------
+    run_start_id: mongoengine.ObjectID
```
Comment (tacaswell) [inline metadatastore/commands.py:505]: This probably should be a string, not an `ObjectID`

Comment (ericdill) [inline metadatastore/commands.py:505]: @arkilic can confirm this, but I believe _either_ will work here?

**Comment by danielballan** [comment]
That's true, but the existing API is wrong. :- D

**Comment by ghost** [comment]
My pr does this. Take a look at PR #105. Find routines are modified per our agreement

File: metadatastore/commands.py (hunk 1 above)
Comment (ghost) [inline metadatastore/commands.py:505]: This works using the method Tom and I have agreed upon earlier:

https://github.com/arkilic/metadatastore/commit/b70253e53318369a3702e600af197be16056be07

**Comment by danielballan** [comment]
@arkilic, I don't think that's true. Your PR does not give us a way to look up Events given a run start ID; it only lets us look up a Events with a RunStart Document that has an `id` attribute. (It looks like that `id` lookup happens in `find_event_descriptor`, which is called by `find_event`.)

If I'm misunderstanding, can you spell out how we should be looking up Events given a run start ID string? Forgive my tired brain.

**Comment by ghost** [comment]
That's the unified way we have agreed upon. We throw run_start objects at all find routines. It was part of the spec. If you are going to use the _id of run start, it should be a bson.ObjectId. This shouldn't be a separate routine, it should be part of find_event. Also, you can always access run_start objects for a given event by eventobj.descriptor.run_start. You don't need have to have a separate routine that does it, mongoengine does it for you using the DbRef field

On Mar 5, 2015, at 7:26 PM, Dan Allan <notifications@github.com<mailto:notifications@github.com>> wrote:

@arkilichttps://github.com/arkilic, I don't think that's true. Your PR does not give us a way to look up Events given a run start ID; I only lets us look up a Events with a RunStart Document that has an id attribute. (It looks like that id lookup happens in find_event_descriptor, which is called by find_event.)

If I'm misunderstanding, can you spell out how we should be looking up Events given a run start ID string? Forgive my tired brain.

**Comment by danielballan** [comment]
@arkilic OK, thanks for reviewing that. The issue we are trying to solve is that DataBroker returns Headers, which are not exactly RunStart objects because they have _two_ IDs, inherited from RunStart and RunEnd, and thus no single `id` attribute. We were thinking we should add a `find_events_from_header` or `find_event_from_run_start_id` to make this work. The alternative -- mutating a Header to look enough like a RunStart to satisfy `find_event` -- was too ugly. What do you think we should do?

**Comment by ghost** [comment]
It's a simple addition to the find_event I put together. I have already done it in an optimized way but with the run_start object as you said. Would you guys be happy if find_event accepted a run_start_id field as string that returned you all the corresponding events for that run_header? I forgot about __as_document goody we added earlier, my bad.

**Comment by danielballan** [comment]
Yes, I think that would do the job.

**Comment by ghost** [comment]
I'll have dinner and add it to my PR.

**Comment by danielballan** [comment]
There's no fire here. It can wait for tomorrow morning.

**Comment by tacaswell** [comment]
Either in #108 or punted to dataportal

**Pull request closed without merging**